nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/random/custom/RandomPoisson.java

```diff
@@ -0,0 +1,62 @@
+package org.nd4j.linalg.api.ops.random.custom;
+
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import org.nd4j.autodiff.samediff.SDVariable;
+import org.nd4j.autodiff.samediff.SameDiff;
+import org.nd4j.base.Preconditions;
+import org.nd4j.imports.descriptors.properties.adapters.DataTypeAdapter;
+import org.nd4j.linalg.api.buffer.DataType;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.api.ops.DynamicCustomOp;
+import org.nd4j.linalg.api.rng.Random;
+import org.tensorflow.framework.AttrValue;
+import org.tensorflow.framework.GraphDef;
+import org.tensorflow.framework.NodeDef;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+@NoArgsConstructor
+public class RandomPoisson extends DynamicCustomOp {
+
+    public RandomPoisson(@NonNull INDArray shape, @NonNull INDArray rate, int... seeds) {
+        addInputArgument(shape, rate);
+        addIArgument(seeds);
+    }
+
+    public RandomPoisson(@NonNull  INDArray shape, @NonNull INDArray rate) {
+        this(shape, rate, 0,0);
+    }
+
+    public RandomPoisson(@NonNull SameDiff sameDiff, @NonNull SDVariable shape, @NonNull SDVariable rate, int... seeds) {
+        super(null, sameDiff, new SDVariable[]{shape, rate});
+        addIArgument(seeds);
+    }
+
+    @Override
+    public String opName() {
+        return "random_poisson";
+    }
+
+    @Override
+    public String[] tensorflowNames() {
+        return new String[]{"RandomPoisson","RandomPoissonV2"};
+    }
+
+    private DataType outputDataType = DataType.FLOAT;
+
+    @Override
+    public void initFromTensorFlow(NodeDef nodeDef, SameDiff initWith, Map<String, AttrValue> attributesForNode, GraphDef graph) {
+        if(attributesForNode.containsKey("rate")) {
+            outputDataType = DataTypeAdapter.dtypeConv(attributesForNode.get("rate").getType());
+        }
+    }
+
+    @Override
+    public List<DataType> calculateOutputDataTypes(List<DataType> inputDataTypes){
+        Preconditions.checkState(inputDataTypes != null, "Expected exactly input datatypes for %s, got null", getClass());
+        return Collections.singletonList(outputDataType);
```

Comment:
Validation message is incomplete - "expected exactly input datatypes"

Reply:
Fixed.

Reply:
Still not right. 
```
Preconditions.checkState(inputDataTypes != null && inputDataTypes.size() == 2, "Expected exactly 2 input datatypes for %s, got null", getClass(), inputDataTypes);
```

Reply:
Fixed.